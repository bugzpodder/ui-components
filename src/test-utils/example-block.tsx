import React from "react";
import styles from "./example-block.module.scss";

type Props = {
  content: any;
  strongHeader: string;
  helperText?: string;
};

export const ExampleBlock = (props: Props) => {
  const { strongHeader, helperText = "", content } = props;
  return (
    <div className={styles.container}>
      <pre data-testid="example" className={styles.text}>
        <strong>{strongHeader}</strong>
        {helperText}= {JSON.stringify(content, null, 2)}
      </pre>
    </div>
  );
};
