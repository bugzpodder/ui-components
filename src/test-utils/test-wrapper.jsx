// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { StyleWrapper } from "../style-wrapper";
import { StylesProvider, createGenerateClassName } from "@material-ui/styles";

type Props = {
  children: Node<*>,
};

// TODO(nsawas): Figure out why this new method of disabling global styles
// isnt exactly working. For now we disable all styles with `disableGeneration` prop below.
const generateClassName = createGenerateClassName({
  disableGlobal: true,
});

class MockResizeObserver {
  observe = () => {};
  disconnect = () => {};
}

window.ResizeObserver = MockResizeObserver;

export const TestWrapper = (props: Props) => {
  const { children } = props;
  return (
    <StylesProvider
      disableGeneration
      generateClassName={generateClassName}
    >
      <StyleWrapper>{children}</StyleWrapper>
    </StylesProvider>
  );
};
