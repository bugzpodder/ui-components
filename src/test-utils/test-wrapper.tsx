import React from "react";
import { StyleWrapper } from "../style-wrapper";
import {
  StylesProvider,
  createGenerateClassName,
} from "@material-ui/core/styles";

// TODO(nsawas): Figure out why this new method of disabling global styles
// isnt exactly working. For now we disable all styles with `disableGeneration` prop below.
const generateClassName = createGenerateClassName({
  disableGlobal: true,
});

class MockResizeObserver {
  observe = (): void => {};
  disconnect = (): void => {};
}

(window as any).ResizeObserver = MockResizeObserver;

export const TestWrapper: React.FC = props => {
  const { children } = props;
  return (
    <StylesProvider disableGeneration generateClassName={generateClassName}>
      <StyleWrapper>{children}</StyleWrapper>
    </StylesProvider>
  );
};
