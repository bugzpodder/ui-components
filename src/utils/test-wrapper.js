// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { createGenerateClassName } from "@material-ui/core/styles";
// $FlowFixMe: Cannot resolve module react-jss/lib/JssProvider
import JssProvider from "react-jss/lib/JssProvider";
import "@grail/lib/src/client-utils/mocks";

type Props = {
	children: Node<*>,
};

const generateClassName = createGenerateClassName({
	dangerouslyUseGlobalCSS: true,
});

export const TestWrapper = (props: Props) => {
	const { children } = props;
	return <JssProvider generateClassName={generateClassName}>{children}</JssProvider>;
};
