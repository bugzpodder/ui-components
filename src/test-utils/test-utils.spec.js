// @flow
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { ExampleBlock } from "./example-block";
import { TestWrapper } from "./test-wrapper";
import { render } from "@testing-library/react";

test("render example block", () => {
  const { container, getByTestId } = render(
    <TestWrapper>
      <ExampleBlock
        strongHeader="State "
        content={{
          test: "one",
        }}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
  expect(getByTestId("example")).toBeInTheDocument();
  expect(getByTestId("example")).toHaveTextContent('State = { "test": "one" }');
});
