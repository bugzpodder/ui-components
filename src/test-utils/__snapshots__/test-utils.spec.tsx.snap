// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render example block 1`] = `
<div>
  <div
    class="container"
  >
    <pre
      class="text"
      data-testid="example"
    >
      <strong>
        State 
      </strong>
      
      = 
      {
  "test": "one"
}
    </pre>
  </div>
</div>
`;
